pkg/apis/core/validation/validation.go

```diff
@@ -3346,6 +3348,13 @@ func ValidatePodSpec(spec *core.PodSpec, podMeta *metav1.ObjectMeta, fldPath *fi
 		}
 	}
 
+	if !opts.AllowNegativeTerminationGracePeriodSeconds && spec.TerminationGracePeriodSeconds != nil {
+		value := *spec.TerminationGracePeriodSeconds
+		if value < 0 || value > math.MaxInt32 {
```

Comment:
this is rejecting values over math.MaxInt32 without considering existing data (the AllowNegativeTerminationGracePeriodSeconds compatibility bit only gets set for negative values)

Reply:
this is still outstanding